### PR TITLE
add JWT token support to client and server

### PR DESCRIPTION
**Pull request by nfstry**
i need the websocket client to include a token for jwt authentication so thought about adding it

### CODE REVIEW

**Comment by pocesar** [comment]
is this directly related to #24? @danielmihai have you started working on the PR or can I merge this, then you rebase?

File: src/server.js [hunk 1]
```diff
@@ -365,7 +379,16 @@ module.exports = function (classes) {
         }
 
         this.authHandler = handler;
+      },
+      enableJWTAuth: function (handler) {
+        /*if (!_.isFunction(handler)) {
```
Comment (danielmihai) [inline src/server.js]: if the handler is not a function, we should throw an exception on setting up the authorization handler, so the server does not boot up with some jibberish handler such as a string.

**Comment by danielmihai** [comment]
I haven't started working on the pull request yet. I think I'll try doing it on Sunday. 

I was thinking of having a simpler approach to setting Authorization. 

The first one is by writing and exposing a general 'set' function for the headers, where the user can set its own headers (which can be marked as an improvement/feature). 

The second one is by having algorithm specific functions for setting the authorization headers, as in `.basic(username, password)` or `.jwt(jwttoken)`. Each one of this approaches does not introduce breaking changes. 

The one that includes breaking changes is by passing an `authOptions` object to the client constructor, in which we specify the auth type (i.e basic, bearer/jwt, digest etc.) together with each auth method parameters (i.e username and password; token etc.) 

Consider one of these suggestions, so I can make a first pull request. 

Thanks,
Daniel :)

**Comment by pocesar** [comment]
hey @nfstry thanks for the PR, we are going to make some big changes for 2.0, and JWT / Bearer is one of them, we will just refactor a good chunk of the code!

**Pull request closed without merging**